src/cpu/execute.rs

```diff
@@ -133,6 +135,10 @@ impl Execute for Mos6502 {
         // ASL
         let (address, cycles) = self.fetch_operand_address(opcode);
         let value = self.read(address);
+
+        if let Mos6502Variant::NMOS = self.variant {
```

Comment:
does it make sense to make this a march statement instead? in case there are other variants we wanna add in the future 👀 :3

Reply:
or maybe thats premature optimization? 🍃

Reply:
wait thoughts on `matches!(thing, pattern)` instead of the if let? probably just a stylistic thing, especially since you dont need to match on any parameters if that makes sense

Reply:
wasn't aware of `matches!()` but seems like exactly what i want! i alternated between `if let` and `==` in this and wasn't rly happy with it

im a bit against a full `match` statement bc that would kind of imply that quirks are likely to have varying behaviors for every version, where these quirks are more just special cases for one version at a time -- e.g. if `NMOS` has a quirk and it was fixed in `CMOS`, it's unlikely to show again up in e.g. the 65816 or something later

Reply:
e.g. you'd never really have a situation like this

```
match version {
  NMOS => do one thing
  CMOS => do another thing
  65816 => do a secret third thing
}
```

its more like, the little `if` statements sprinkled throughout account for each chip's separate set of errata, so something special-cased in for an NMOS quirk but skipped on CMOS wouldn't really be applicable if we added in a third generation

Reply:
yeep, my 🍃 ass only realized that after i left the comment, makes total sense :3